docs/end-user/policies/gc.md

```diff
@@ -0,0 +1,254 @@
+---
+title: Garbage Collect
+---
+
+By default, KubeVela Application will recycle outdated resources when new version is deployed and confirmed to be healthy. In some cases, you may want to have more customized control to the recycle of outdated resources, where you can leverage the garbage-collect policy.
+
+In garbage-collect policy, there are two major capabilities you can use.
+
+## Keep legacy resources
+
+Suppose you want to keep the resources created by the old version of the application. Use the garbage-collect policy and enable the option `keepLegacyResource`.
+
+```yaml
+# app.yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: first-vela-app
+spec:
+  components:
+    - name: express-server
+      type: webservice
+      properties:
+        image: crccheck/hello-world
+        port: 8000
+      traits:
+        - type: ingress-1-20
+          properties:
+            domain: testsvc.example.com
+            http:
+              "/": 8000
+  policies:
+    - name: keep-legacy-resource
+      type: garbage-collect
+      properties:
+        keepLegacyResource: true
+```
+
+1. create app
+
+``` shell
+vela up -f app.yaml
+```
+
+```shell
+$ kubectl get app
+NAME             COMPONENT        TYPE         PHASE     HEALTHY   STATUS   AGE
+first-vela-app   express-server   webservice   running   true               29s
+```
+
+2. update the app
+
+```yaml
+# app1.yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: first-vela-app
+spec:
+  components:
+    - name: express-server-1
+      type: webservice
+      properties:
+        image: crccheck/hello-world
+        port: 8000
+      traits:
+        - type: ingress-1-20
+          properties:
+            domain: testsvc.example.com
+            http:
+              "/": 8000
+  policies:
+    - name: keep-legacy-resource
+      type: garbage-collect
+      properties:
+        keepLegacyResource: true
+```
+
+``` shell
+vela up -f app1.yaml
+```
+
+```shell
+$ kubectl get app
+NAME             COMPONENT          TYPE         PHASE     HEALTHY   STATUS   AGE
+first-vela-app   express-server-1   webservice   running   true               9m35s
+```
+
+check whether legacy resources are reserved.
```

Comment:
can we use `vela status` to check?

Or, let's add some explanation like:

```
In the following steps, we'll use `kubectl` command to do some verification.
```

Reply:
Ok, I will add some explanation. The `vela status` can see the status of the components but kubectl can verify the resources more directly.

Reply:
it seems not added here yet?

Reply:
Added above.